packages/grid/_modules_/grid/components/cell/GridCell.tsx

```diff
@@ -128,10 +128,14 @@ export const GridCell: React.FC<GridCellProps> = React.memo((props) => {
   };
 
   React.useLayoutEffect(() => {
-    if (hasFocus && cellMode === 'view' && cellRef.current) {
+    if (
+      hasFocus &&
+      cellRef.current &&
+      (!document.activeElement || !cellRef.current!.contains(document.activeElement))
+    ) {
       cellRef.current!.focus();
     }
-  });
+  }, [hasFocus]);
```

Comment:
fix #1295, basically we trigger focus only when hasFocus changed

Reply:
This is not what I understand of the problem, this seems to be a symptom, not the root cause. From what I understand we should filter all the events that are coming from a portal in 

https://github.com/mui-org/material-ui-x/blob/728bad3ef1f481a84a042d9bb44da970b56b5c87/packages/grid/_modules_/grid/components/GridRow.tsx#L35

```diff
diff --git a/packages/grid/_modules_/grid/components/GridRow.tsx b/packages/grid/_modules_/grid/components/GridRow.tsx
index 27135564..f1a3f0be 100644
--- a/packages/grid/_modules_/grid/components/GridRow.tsx
+++ b/packages/grid/_modules_/grid/components/GridRow.tsx
@@ -27,8 +27,14 @@ export const GridRow: React.FC<RowProps> = ({ selected, id, className, rowIndex,
   const rowHeight = useGridSelector(apiRef, gridDensityRowHeightSelector);

   const publish = React.useCallback(
-    (eventName: string) => (event: React.MouseEvent) =>
+    (eventName: string) => (event: React.MouseEvent) => {
+      // Ignore portal
+      if (!event.currentTarget.contains(event.target)) {
+        return;
+      }
+
       apiRef!.current.publishEvent(eventName, apiRef?.current.getRowParams(id), event),
+    },
     [apiRef, id],
   );
```

what do you think?

Reply:
Why would a redraw impact the focus? Maybe you are referring to what happens when we toggle between the read and edit mode, in which case, maybe we can only trigger a focus when the edit/read mode change?